Lib/functools.py

```diff
@@ -234,9 +234,9 @@ def __ge__(self, other):
 
 _initial_missing = object()
 
-def reduce(function, sequence, initial=_initial_missing):
+def reduce(function, iterable, initial=_initial_missing, /):
```

Comment:
This is a breaking change.

Reply:
No, it's not. The Python implementation is replaced with the C implementation:

```python
def reduce(function, iterable, initial=_initial_missing, /):
    ...

try:
    from _functools import reduce
except ImportError:
    pass
```

which accepts no keyword argument.

The import after the function definition always succeeds in CPython. This may be a breaking change for other Python implementations, such as PyPy.

```python
In [1]: import functools

In [2]: type(functools.reduce)
Out[2]: builtin_function_or_method

In [3]: functools.reduce(lambda x, y: x + y, [1, 2, 3, 4])
Out[3]: 10

In [4]: functools.reduce(lambda x, y: x + y, sequence=[1, 2, 3, 4])
TypeError: reduce() takes no keyword arguments

In [5]: functools.reduce(lambda x, y: x + y, iterable=[1, 2, 3, 4])
TypeError: reduce() takes no keyword arguments
```

Reply:
python implementation is also used by other projects, like pypy.
to, this is a breaking change.

Reply:
> python implementation is also used by other projects

That's fine. I revert the signature change here.